### ISSUE TITLE
Condition not recognized

### ISSUE THREAD

**Issue by Francky63** [WhiteMagic/JoystickGremlin#150]
the condition "Joystick Button 3 is pressed" in map keybord is not recognized evry time i run Gremlin the profile.

Every time i have to reactivated it and save but when i reload again its agai not recognized...
What is strange is that the others buttons in remap for the same devices are regnized.

**Comment by Francky63** [comment]
i don't know if is related to #147

**Comment by WhiteMagic** [comment]
Hm that's odd, are other conditions working fine? One reason could be that Gremlin records a windows ID for the condition which maybe changes. I'm currently looking into making this entire joystick identification a bit saner and less brittle.

**Comment by Francky63** [comment]
yes as reported in #147 i have to connect device by device and it happens that the id is changed but as as i saw in the normal device setup is automatically updated but not in condition.
I forget to attach an example 

[DCS.xml.txt](https://github.com/WhiteMagic/JoystickGremlin/files/2492945/DCS.xml.txt)

As you can see at row 35 the windows_id is 2, at row 69 the windows_id is 4 for the same device

**Comment by WhiteMagic** [comment]
I had a quick look at the code and that code works off the windows id, so if that changes the condition will read from a stick you didn't specify initially. While there isn't really anything to work around this at with the current version of Gremlin I'm at the moment redoing the whole device identification thing to be less awkward and capable of updating itself.

The conditions will be a bit tricky as they'd have to be updated upon loading the profile as they save the hardware and windows id and don't care when those change. So I have to update the windows id at load time as long as the hardware id stays unique. In case of duplicate devices the hardware id would no longer be unique and the windows ids would be the only distinguishing factor.
